feat(core/toast): Added pause api to toast

1307-Dev opened the pull request:
## 💡 What is the current behavior?

There is currently no explicit method/api for pausing a toast

GitHub Issue Number: #<ISSUE NUMBER>
Jira- IX-2213

## 🆕 What is the new behavior?

Have added an explicit method for pausing the toast. This method can be reused when the UI designs are updated.

## 🏁 Checklist

A pull request can only be merged if all of these conditions are met (where applicable):

- [ ] 🦮 Accessibility (a11y) features were implemented
- [ ] 🗺️ Internationalization (i18n) - no hard coded strings
- [ ] 📲 Responsiveness - components handle viewport changes and content overflow gracefully
- [ ] 📕 Add or update a Storybook story
- [ ] 📄 Documentation was reviewed/updated [siemens/ix-docs](https://github.com/siemens/ix-docs)
- [x] 🧪 Unit tests were added/updated and pass (`pnpm test`)
- [ ] 📸 Visual regression tests were added/updated and pass ([Guide](https://github.com/siemens/ix/blob/main/CONTRIBUTING.md#visual-regression-testing))
- [x] 🧐 Static code analysis passes (`pnpm lint`)
- [x] 🏗️ Successful compilation (`pnpm build`, changes pushed)

## 👨‍💻 Help & support

1307-Dev commented on packages/core/src/components/toast/toast-container.tsx:
```diff
@@ -117,6 +117,16 @@ export class ToastContainer {
+        if (typeof (toast as any).resume === 'function') {
+          (toast as any).resume();
+        }
+      },
```
Removed the casts

1307-Dev commented on packages/core/src/components/toast/toast.tsx:
```diff
@@ -32,6 +33,15 @@ import {
+  @Method()
+  async resume() {
+    this.pausedByApi = false;
+  }
```
Repositioned the methods

danielleroux commented on packages/core/src/components/toast/toast-container.tsx:
```diff
@@ -117,6 +117,16 @@ export class ToastContainer {
         removeToast(result);
       },
+      pause: () => {
+        if (typeof toast.pause === 'function') {
+          toast.pause();
+        }
+      },
+      resume: () => {
+        if (typeof toast.resume === 'function') {
+          toast.resume();
+        }
```
typeof check is not necessary, its already a known type HTMLIxToastElement

danielleroux commented on packages/core/src/components/toast/toast.tsx:
```diff
@@ -147,14 +149,24 @@ export class Toast {
     }, 250);
   }
 
+  @Method()
+  async pause() {
+    this.paused = true;
+  }
+
+  @Method()
+  async resume() {
+    this.paused = false;
+  }
```
Function needs jsdocs these are missing here

danielleroux reviewed (changes requested):
Currently its not possible to get any information if the toast is paused.

I would suggest add a additional function to the toast which returns the current state of the toast

1307-Dev commented:
> Currently its not possible to get any information if the toast is paused.
> 
> I would suggest add a additional function to the toast which returns the current state of the toast

I have added another method called isPaused() which returns a boolean based on whether the toast is paused. I have added tests for the same.

danielleroux commented on packages/core/src/components/toast/toast.tsx:
```diff
@@ -147,14 +149,38 @@ export class Toast {
+    this.paused = false;
+  }
+
+  /**
+   * Returns whether the toast is currently paused (auto-close is paused).
+   */
+  @Method()
+  async isPaused(): Promise<boolean> {
+    return this.paused;
+  }
```
How about the following case taking in the "touched"
```suggestion
  /**
   * Returns whether the toast is currently paused (auto-close is paused).
   */
  @Method()
  async isPaused(): Promise<boolean> {
    return this.paused || this.touched;
  }
```
